### PR TITLE
Bump the version of `go-restful` to 2.16.0

### PR DESCRIPTION
**Pull request by Barakmor1**
Signed-off-by: bmordeha <bmodeha@redhat.com>

**What this PR does / why we need it**:
Bump the version of emicklei/go-restful from 2.15.0  to 2.16.0 to fix https://github.com/emicklei/go-restful/pull/493
See https://github.com/emicklei/go-restful/issues/489#issuecomment-1148861521.
(https://github.com/emicklei/go-restful/pull/503)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
**Modified files ( not auto generated ) :**
go.mod
staging/src/kubevirt.io/client-go/go.mod

**Release note**:

```release-note
Bump the version of emicklei/go-restful from 2.15.0 to 2.16.0
```

### CODE REVIEW

File: staging/src/kubevirt.io/client-go/go.mod [hunk 1]
```diff
@@ -71,6 +71,7 @@ require (
 )
 
 replace (
+	github.com/emicklei/go-restful => github.com/emicklei/go-restful v2.16.0+incompatible
```
[the remark this answers is not in the document]
Comment (Barakmor1) [reply staging/src/kubevirt.io/client-go/go.mod:74]: This cause that dependencies that use `go-restful` to use  v2.16.0
without it `go-restful-openapi` and other libraries use old versions of `go-restful`

found the solutions in here:
https://stackoverflow.com/questions/69825533/why-does-go-sum-include-so-many-older-packages

[the remark this answers is not in the document]
Comment (xpivarc) [reply staging/src/kubevirt.io/client-go/go.mod:74]: I went through your references but I am not sure what I should focus on. Can you be more specific? I think what you are looking for and what will contradict is how Go selects the library version it should use ([ref](https://go.dev/ref/mod#minimal-version-selection) )

[the remark this answers is not in the document]
Comment (Barakmor1) [reply staging/src/kubevirt.io/client-go/go.mod:74]: Thanks! 
Will update soon

**Review by xpivarc — COMMENTED** [review]
/approve

**Comment by iholder-redhat** [comment]
> I would prefer if the git log history is populated with meaningful commit messages. Please consider adding the details of why this change is done to the commit message as well.

It's a bit off-topic for this PR :) but I'm having hard time to understand why this matters.

PR description is way more readable, can update after the PR merges, can contain images, proper links and so on. How would we benefit to duplicate this description into the commit messages? Especially when the PR description updates and the commit message doesn't (this can also happen before the PR merges - it's easy to remember to update the PR and forget updating the commit messages). Also, when you `git log`, you would easily find the PR number and can search for it and find the description.

I know it's an unpopular opinion, but I honestly don't think that saving a few seconds of searching the PR number in github is worth duplicating description that later can contradict or at least be different then what's explained at the PR.

**Comment by Barakmor1** [comment]
@EdDev Done.

**Comment by dhiller** [comment]
/hold for getting the clone fix PR in

FYI @brianmcarey

**Pull request merged**